### PR TITLE
Migration of cloud.gov home and sub pages, get started, contact, news listing pages, knowledge base.

### PR DESCRIPTION
**Pull request by jamesthebrooks**
## Changes proposed in this pull request:

- Why [cloud.gov](http://cloud.gov/) and sub pages. 1:1 from jekyll. Ready for complete review.
- Get Started 1:1 from jekyll. Ready for complete review.
- Contact us 1:1 from jekyll. Ready for complete review.
- News and sub pages. Content has been migrated, but not reviewed. Expect missing images.
- Knowledge base and sub pages. Content has been migrated, but not reviewed. Expect missing images.
- Setup and implementation of anchorjs, mermaidjs, and prismjs.
- Helper filters added such as "debug" and "markdown" that can be used in templates to test objects
- Custom url rewriting to match url structure in the jekyll site
- eleventy-plugin-toc set up and implemented to handle table of contents within templates
- wrote python script to handle invalid dates in md front matter. it's in the root. will add instructions later

Ignore the sidenav. Children don't behave yet.
Ignore anything not explicitly listed in this or prior peer reviews as this is part of a series up updates for this migration epic.

### CODE REVIEW

**Comment by apburnes** [comment]
@jamesthebrooks does #2 need to be merged before this one?

File: .eleventy.js [hunk 1]
```diff
@@ -7,16 +7,34 @@ const markdownItAnchor = require('markdown-it-anchor');
...
   let pathPrefix = '/';
 
   // Copy the `admin` folders to the output
-  config.addPassthroughCopy('admin');
+  // config.addPassthroughCopy('admin');
 
   // Copy USWDS init JS so we can load it in HEAD to prevent banner flashing
-  config.addPassthroughCopy({'./node_modules/@uswds/uswds/dist/js/uswds-init.js': 'assets/js/uswds-init.js'});
+  config.addPassthroughCopy({
+    'admin': 'admin',
```
Comment (sknep) [inline .eleventy.js:26]: admin is used for netlify cms, you probably don't need it for this build

File: .eleventy.js [hunk 2]
```diff
@@ -36,9 +54,29 @@ module.exports = function (config) {
...
+  config.addPlugin(syntaxHighlight, {
+
+  });
+
+  config.addPlugin(pluginTOC, {
+    tags: ['h2']
+  });
+
+  config.addPlugin(pluginMermaid, {
+    mermaid_js_src: '/assets/mermaid.min.js',
```
Comment (sknep) [inline .eleventy.js:66]: is mermaid loading on every page? i don't know how often it is used across the site, but doesn't seem like all that often

File: .eleventy.js [hunk 3]
```diff
@@ -134,6 +182,22 @@ module.exports = function (config) {
...
+    }
+    if (page.url.startsWith("/pages/news/articles/")) {
+      let dateMatches = page.url.match(/[0-9]{4}-[0-9]{2}-[0-9]{2}/g);
+      let date = !!dateMatches && dateMatches.length > 0 ? dateMatches[0].replaceAll('-','/') : '';
+      let descriptionMatches = page.url.match(/[A-Za-z]+-[A-Za-z]+/);
+      let description = !!descriptionMatches && descriptionMatches.length > 0 ? descriptionMatches[descriptionMatches.length - 1] + '/' : '';
+      return `/${date}/${description}`;
+    }
+  });
+
```
Comment (sknep) [inline .eleventy.js:200]: I think these might be better done using folder-level defaults values in the directory data files in eleventy: https://www.11ty.dev/docs/data-template-dir/

File: _data/pages/navigation.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    href: /pages/pricing/
+  - text: Knowledge Base
+    href: /pages/knowledge-base/
+  - text: Contact
+    href: /pages/contact/
+  - text: Manage Sites
+    href: https://pages.cloud.gov
+    external: true
+
+sidenav:
```
Comment (sknep) [inline _data/pages/navigation.yml:42]: is there no way to do this dynamically? this won't be up to date as soon as someone creates a new page

File: _data/site.yaml [hunk 5]
```diff
@@ -41,64 +55,300 @@ twitter: 18F
...
+  - Gemfile.lock
+  - .parcel-cache
+
+### plugin settings ###
+last-modified-at:
+  date-format: "%Y-%m-%d"
+
+
+defaults:
+  - scope:
```
Comment (sknep) [inline _data/site.yaml:220]: I think this whole list could be just one item:
```
  - scope: 
      path: "pages/docs"
    values:
      toc: true
```
and get the same effect.

**Comment by sknep** [comment]
One idea for making this PR slightly more manageable to review -- remove the images from the history, and set up a separate PR for them. I think they're just tanking the GitHub interface. Just an idea.

I would love to see these commit message with a little more structure. We use conventional commits for the Pages repo and it really helps. Getting used to rebasing is something I picked up after being here for a bit, but I'm now a convert.

**Pull request closed without merging**